supervisor/utils/__init__.py

```diff
@@ -39,20 +39,21 @@ async def wrap_api(api, *args, **kwargs):
     return wrap_api
 
 
-def check_port(address: IPv4Address, port: int) -> bool:
+async def async_check_port(
```

Comment:
We don't have to change the name of these functions just because we made them `async`.

It's also a bit confusing to do so actually. Core has a standard practice to prefix names of utility methods with `async` when they must be called in the event loop. That doesn't always mean they return a coroutine, non-async methods that have to be called in the event loop follow this naming convention as well, see https://github.com/home-assistant/core/blob/cd9ad32e329a2106b303ed30c661d668347ce02d/homeassistant/config_entries.py#L767 for example.

Supervisor hasn't been following this convention. We can but we have a lot of refactoring to do if we want to. I don't want to introduce this prefix randomly in a few places and confuse people who are coming over from core. We should either adopt the same convention in a refactoring PR or continue to not use it.

Reply:
> We don't have to change the name of these functions just because we made them `async`.
> 
> It's also a bit confusing to do so actually. Core has a standard practice to prefix names of utility methods with `async` when they must be called in the event loop. That doesn't always mean they return a coroutine, non-async methods that have to be called in the event loop follow this naming convention as well, see https://github.com/home-assistant/core/blob/cd9ad32e329a2106b303ed30c661d668347ce02d/homeassistant/config_entries.py#L767 for example.
> 
> Supervisor hasn't been following this convention. We can but we have a lot of refactoring to do if we want to. I don't want to introduce this prefix randomly in a few places and confuse people who are coming over from core. We should either adopt the same convention in a refactoring PR or continue to not use it.

The supervisor is async per default, if a function is not, they have it normally written in the function description